tests/typing/pyrightconfig.json

```diff
@@ -0,0 +1,3 @@
+{
```

Comment:
Move this to `pyproject.toml`.

Reply:
As in, create a `pyproject.toml` in `tests/typing`? No `pyproject.toml` in the root of the 8.1.x branch this PR is targetting.

Reply:
If you meant the latter, I think it might give people to idea that click can be type checked with Pyright (as opposed to its public types).  Also note that Pyright's TOML parser does not support v1 of the spec (https://github.com/microsoft/pyright/issues/4336), meaning that you won't be able to use v1 constructs in your `pyproject.toml`.

Reply:
That's fine, just leave it as is for this branch.